erpnext/www/lms/course.py

```diff
@@ -20,7 +20,19 @@ def get_context(context):
 	context.topics = course.get_topics()
 	context.has_access =  utils.allowed_program_access(context.program)
 	context.progress = get_topic_progress(context.topics, course, context.program)
+	context.total_progress = calculate_progress(context.topics,context.progress)
+
 
 def get_topic_progress(topics, course, program):
 	progress = {topic.name: utils.get_topic_progress(topic, course.name, program) for topic in topics}
 	return progress
+
+def calculate_progress(topics,progress):
+	total_topic = len(topics)
+	complete_topic = 0
+	for p in progress:
+		for status in progress[p]:
+			if status == "completed" and progress[p][status] == True:
```

Comment:
```suggestion
	for p in progress:
			if p.get("completed"):
```

Reply:
@meshramsaroj  instead of p add some naming convention like 
for progress in progresses:

Reply:
while writing code make sure you get an idea about time complexity here your chunk of code required O(n^2) complexity, it can easily solve in O(n) complexity.

Reply:
fixed this changes